### ISSUE TITLE
vers 3.7.0: plt.fill_between; set_xlim(date)

### ISSUE THREAD

**Issue by Dibal** [matplotlib/matplotlib#25257]
There are at least two bugs; I solved the problem by going back to matplotlib-3.6.3:

* Restricting plotting limits by "ax.set_xlim('2022-01-01', ...)" does not work any more. Perhaps matplotlib cannot convert units any more.
* "plt.fill_between" does not work. Area is shown in legend correctly but not inside the plot.

**Comment by jklymak** [comment]
Please provide a full example including imports.  Strings never worked in Matplotlib so strongly suspect you are using pandas.

**Comment by Dibal** [comment]
You are right, my example used Pandas data. But the error messages came from matplotlib (and datetime('2023,1,1') did not work either). Downgrading to 3.6.3. healed the problems. If I am the first one who encountered the problem  I'll prepare a demo soon.

**Comment by ksunden** [comment]
For pandas and xlim see #25219 for an extensive discussion. We have multiple possible fixes, but are deciding on which one(s) to apply.

For fill between, I really can't say without an example.

**Comment by ZupoLlask** [comment]
There's also issues with axvspan and axvline, when used with pandas.

**Comment by ksunden** [comment]
There's a good chance those are related, and will be resolved with the same solution

**Comment by ZupoLlask** [comment]
I guess so... 

I'll try to post reductions of the issues I'm seeing.

**Comment by Dibal** [comment]
Thanks! The linked thead is exactly the error I got. I fail to reproduce "fill_between" error based on simple dataset. On my complex working dataset the hatch-filling is simply missing, without error message. It reappears after switching to matplotlib-3.6.3.

**Comment by ZupoLlask** [comment]
@ksunden @jklymak
Just wanted to let you know that while trying to prepare a reduction of my issues, I came to the conclusion that my issue is related with an incompatibility between Pandas 1.5.3 plot method and Matplotlib 3.7.0 as backend.

Under certain circumstances, np.datetime64 based x axis (index) is ploting data starting from 1970 event if there's no such date in the dataset.

I'll report the issue there.
